WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesViewController.swift

```diff
@@ -90,4 +98,9 @@ private extension ProductImagesViewController {
     @objc func completeEditing() {
         onCompletion(productImages)
     }
+
+    func onDeletion(productImage: ProductImage) {
+        productImages.removeAll(where: { $0.imageID == productImage.imageID })
```

Comment:
Nit (feel free to ignore it): you can also write in a more compact way `productImages.removeAll{ $0.imageID == productImage.imageID }`